build.gradle

```diff
@@ -1,32 +1,57 @@
 plugins {
     id 'java'
     id 'checkstyle'
+    id 'idea'
     id 'application'
-    id 'com.github.johnrengelman.shadow' version '2.0.3'
+    id 'com.github.johnrengelman.shadow' version '2.0.4'
     id 'com.liferay.node' version '4.4.0'
 }
+
 mainClassName = 'reposense.RepoSense'
+
 node.nodeVersion = '10.5.0'
 
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
 repositories {
     mavenCentral()
 }
 
+wrapper {
+  gradleVersion = '4.9'
+}
+
 checkstyle {
     toolVersion = '8.1'
 }
 
 test {
-  testLogging.showStandardStreams = true
+  testLogging {
+    events 'passed', 'skipped', 'failed'
+    showStandardStreams = true
+  }
+}
+
+sourceSets {
+  functional {
+    compileClasspath += main.output + test.output
+    runtimeClasspath += main.output + test.output
+    java.srcDir file('src/functional/java')
+    resources.srcDir file('src/functional/resources')
+  }
+}
+
+configurations {
+  functionalImplementation.extendsFrom testImplementation
+  functionalRuntime.extendsFrom testRuntime
 }
 
 dependencies {
-    implementation  group: 'commons-collections', name: 'commons-collections', version: '3.2'
-    implementation  'com.github.javaparser:javaparser-core:3.0.1'
-    implementation  'com.google.code.gson:gson:2.6.2'
-    implementation  group: 'org.json', name: 'json', version: '20180130'
-    implementation  group: 'net.sourceforge.argparse4j', name: 'argparse4j', version: '0.8.1'
+    implementation  group: 'com.github.javaparser', name: 'javaparser-core', version: '3.0.1'
+    implementation  group: 'com.google.code.gson' ,name: 'gson', version:'2.6.2'
```

Comment:
Should we also change to the latest `gson` version `2.8.2`?

https://github.com/google/gson/blob/master/CHANGELOG.md
**2.6.2** `2016-02-26`
**2.8.5** `2018-05-21`